02_Util/src/util/swagger.ts

```diff
@@ -117,4 +125,126 @@ export function initSwagger(
+  headers: string[],
+  key: string,
+): string | undefined => {
+  for (let i = 0; i < headers.length; i += 2) {
+    if (headers[i].toLowerCase() === key.toLowerCase()) {
+      return headers[i + 1];
+    }
+  }
+  return undefined;
+};
+
+const getTokenFromAuthHeader = (
+  authorizationHeader: string | undefined,
+): string | undefined => {
+  if (!!authorizationHeader) {
+    const token = authorizationHeader.split('Bearer ')[1];
+    return token;
+  }
+  return undefined;
+};
+
+const getAuthorizationTokenFromRawHeaders = (
+  headers: string[],
+): string | undefined => {
+  const authorizationHeader = getHeaderValue(headers, HttpHeader.Authorization);
+  return getTokenFromAuthHeader(authorizationHeader);
+};
+
+export const getAuthorizationTokenFromRequest = (
+  request: FastifyRequest,
+): string => {
+  const token = getAuthorizationTokenFromRawHeaders(request.raw.rawHeaders);
+  if (!token) {
+    throw new UnauthorizedException('Token not found in headers');
+  }
+  return token;
+};
+
+const registerFastifyAuth = async (server: FastifyInstance) => {
+  await server.register(FastifyAuth as any).after();
+  console.log((server as any).authorization);
+
+  server.decorate(
+    'authorization',
+    function (request: any, reply: any, done: any) {
+      try {
+        const token = getAuthorizationTokenFromRequest(request);
+        if (token === '123') {
+          // todo managing real auth but for now using hard coded token
+          done(); // pass an error if the authentication fails
+        } else {
+          throw new UnauthorizedException('Token not authorized');
+        }
+      } catch (e) {
+        reply
+          .code(HttpStatus.UNAUTHORIZED)
+          .send(OcpiErrorResponse.build(HttpStatus.UNAUTHORIZED));
+      }
+    },
+  );
+};
+
+const registerFastifySwagger = (
+  systemConfig: SystemConfig,
+  server: FastifyInstance,
+) => {
+  server.register(fastifySwagger, {
+    openapi: {
+      info: {
+        title: 'CitrineOS Central System API',
+        description: 'Central System API for OCPP 2.0.1 messaging.',
+        version: '1.1.1',
+      },
+      servers: [
+        {
+          url: `http://${systemConfig.centralSystem.host}:${systemConfig.centralSystem.port}`,
+          description: 'TODO',
+        },
+      ],
+      components: {
+        securitySchemes: {
+          authorization: {
+            type: 'http',
+            scheme: 'bearer',
+          },
+        },
+      },
+      tags: Object.values(OcpiTag).map((tag) => {
+        return {
+          name: getOcpiTagString(tag),
+          description: `OCPI ${getOcpiTagString(tag)} endpoints`,
+        };
+      }),
+    },
+    transformObject: OcppTransformObject,
+  });
+};
+
+export async function initSwagger(
+  systemConfig: SystemConfig,
+  server: FastifyInstance,
+) {
+  registerFastifySwagger(systemConfig, server);
+  registerSwaggerUi(systemConfig, server);
+  await registerFastifyAuth(server);
```

Comment:
adjusted previous implementation to handle the
```
await server.register(FastifyAuth as any).after();
```
await because otherwise the `authorization` and `.auth` are not available when creating the endpoints in `AbstractModuleApi`